packages/react-vis/src/treemap/index.js

```diff
@@ -91,30 +91,18 @@ function _getScaleFns(props) {
   };
 }
 
-class Treemap extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      scales: _getScaleFns(props),
-      ...getInnerDimensions(props, props.margin)
-    };
-  }
-
-  UNSAFE_componentWillReceiveProps(props) {
-    this.setState({
-      scales: _getScaleFns(props),
-      ...getInnerDimensions(props, props.margin)
-    });
-  }
+function Treemap(props) {
+  const scales = _getScaleFns(props);
+  const innerDimensions = getInnerDimensions(props, props.margin);
```

Comment:
May want to memoize these as well, but it may not matter too much since I believe that they are fairly small methods.

Reply:
I think memoization makes sense your previous comment as the computation looks a bit complex, but I am more reticent to use memoization on `getInnerDimensions`

My rule of thumb for knowing whether to use memoization or not is "do not use it first, and if you have perf issues, try to fix it with memoization". Otherwise you can in some cases do more harm than good since memoization does have its tradeoffs:

see  [this link](https://blog.logrocket.com/rethinking-hooks-memoization/) and also [that one](https://kentcdodds.com/blog/usememo-and-usecallback)

Reply:
True, memoization does have its trade offs, but I've also seen the "death by a thousand cuts" way too many times before.  Usually its the culmination of a whole bunch of little performance hits that don't look very big on their own.
Its more of a concern in a bigger app, rather than a library, but once you get to that point its hard to get out of.

I'm not too worried about`getInnerDimensions` since it is simple and is passed as primatives, however `_getScaleFns` creates new functions every execution.  Although this would be better fixed at a higher level (as per my other PR).